pkg/cmd/codespace/ports.go

```diff
@@ -107,40 +107,42 @@ func (a *App) ListPorts(ctx context.Context, selector *CodespaceSelector, export
 	}
 
 	cs := a.io.ColorScheme()
-	//nolint:staticcheck // SA1019: utils.NewTablePrinter is deprecated: use internal/tableprinter
-	tp := utils.NewTablePrinter(a.io)
-
-	if tp.IsTTY() {
-		tp.AddField("LABEL", nil, nil)
-		tp.AddField("PORT", nil, nil)
-		tp.AddField("VISIBILITY", nil, nil)
-		tp.AddField("BROWSE URL", nil, nil)
+	tp := tableprinter.New(a.io)
+
+	if a.io.IsStdoutTTY() {
+		tp.AddField("LABEL")
+		tp.AddField("PORT")
+		tp.AddField("VISIBILITY")
+		tp.AddField("BROWSE URL")
 		tp.EndRow()
 	}
 
 	for _, port := range portInfos {
-		tp.AddField(port.Label(), nil, nil)
-		tp.AddField(strconv.Itoa(port.SourcePort), nil, cs.Yellow)
-		tp.AddField(port.Privacy, nil, nil)
-		tp.AddField(port.BrowseURL(), nil, nil)
+		// Convert the ACE to a friendly visibility string (private, org, public)
+		visibility := portforwarder.AccessControlEntriesToVisibility(port.Port.AccessControl.Entries)
+
+		tp.AddField(port.Label())
+		tp.AddField(cs.Yellow(strconv.Itoa(int(port.Port.PortNumber))))
```

Comment:
I dont know what the type of `port.Port.PortNumber` is but it feels real weird to convert it to an int and then to a string... Maybe there is something simpler we can do?

Reply:
It's a `uint16` which is the reason for the int->string conversion, but I went ahead and updated it to use `Sprintf` to do the conversion for us 🙂